### PR TITLE
Document SO_REUSEADDR to be the default for TCP `bind`

### PR DESCRIPTION
**Pull request by badeend**
For TCP sockets specifically, this is a reasonable default for WASI to recommend:
- For the vast majority of cases, this is "what you want"
- On Windows this is the default behavior and can't be opted out of. In other words: every existing cross-platform application already has to deal with this behavior.
- This makes it easier to implement wasi-sockets on top of standard libraries that set SO_REUSEADDR by default too. Examples:
  - https://github.com/rust-lang/rust/blob/e51e98dde6a60637b6a71b8105245b629ac3fe77/library/std/src/sys_common/net.rs#L408
  - https://github.com/libuv/libuv/blob/v1.x/src/unix/tcp.c#L167
  - https://github.com/dotnet/runtime/blob/23e5d5410ed3e8bc9ab61b3685e1d3a06bdba701/src/native/libs/System.Native/pal_networking.c#L1566
  - https://github.com/golang/go/blob/master/src/net/sockopt_linux.go#L28
- This does not preclude us from adding an advanced option in the future to opt-out of this.

---

This is the behavior as implemented by [Wasmtime](https://github.com/bytecodealliance/wasmtime/blob/8573214213c05f970f29a998016f190c3766a789/crates/wasi/src/preview2/host/tcp.rs#L57) and [JCO](https://github.com/libuv/libuv/blob/v1.x/src/unix/tcp.c#L167).

### CODE REVIEW

**Comment by badeend** [comment]
@sunfishcode Do you agree?

File: imports.md [hunk 1]
```diff
@@ -1190,6 +1190,10 @@ implicitly bind the socket.</p>
 <li><code>not-in-progress</code>:           A <code>bind</code> operation is not in progress.</li>
 <li><code>would-block</code>:               Can't finish the operation, it is still in progress. (EWOULDBLOCK, EAGAIN)</li>
 </ul>
+<h1>Implementors note</h1>
+<p>When binding to a non-zero port, this bind operation shouldn't be affected by the TIME_WAIT
+state of a recently closed socket on the same local address (i.e. the SO_REUSEADDR socket
+option should be set implicitly on platforms that require it).</p>
```
Comment (sunfishcode) [inline imports.md:1196]: It looks like Windows' `SO_REUSEADDR` has [different and undesirable behavior](https://docs.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse) so it'd be good to call out that this advice doesn't apply on Windows.

**Review by sunfishcode — APPROVED** [review]
This sounds reasonable to me. If we encounter a use case where someone specifically wants `SO_REUSEADDR` disabled, we can evolve by adding methods to explicitly set/get that flag (which might just fail on Windows).

File: imports.md (hunk 1 above)
Comment (badeend) [reply imports.md]: Done!

**Pull request merged**